entsoe-balances-adjustment/src/main/java/com/powsybl/entsoe/balances_adjustment/util/CgmesVoltageLevelsArea.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.entsoe.balances_adjustment.util;
+
+import com.powsybl.balances_adjustment.util.NetworkArea;
+import com.powsybl.cgmes.extensions.CgmesControlArea;
+import com.powsybl.cgmes.extensions.CgmesDanglingLineBoundaryNode;
+import com.powsybl.cgmes.extensions.CgmesLineBoundaryNode;
+import com.powsybl.iidm.network.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+class CgmesVoltageLevelsArea implements NetworkArea {
+
+    private final List<String> voltageLevelIds = new ArrayList<>();
+
+    private final List<DanglingLine> danglingLineBordersCache;
+    private final List<Branch<?>> branchBordersCache;
+
+    private final Set<Bus> busesCache;
+
+    CgmesVoltageLevelsArea(Network network, CgmesControlArea area, List<String> excludedXnodes, List<String> voltageLevelIds) {
+        this.voltageLevelIds.addAll(voltageLevelIds);
+
+        danglingLineBordersCache = createDanglingLinesCache(network, area, excludedXnodes);
+        branchBordersCache = createBranchesCache(network, area, excludedXnodes);
+
+        busesCache = network.getBusView().getBusStream()
+                .filter(bus -> voltageLevelIds.contains(bus.getVoltageLevel().getId()))
+                .filter(Bus::isInMainSynchronousComponent)
+                .collect(Collectors.toSet());
+    }
+
+    private List<DanglingLine> createDanglingLinesCache(Network network, CgmesControlArea area, List<String> excludedXnodes) {
+        return network.getDanglingLineStream()
+                .filter(this::isAreaBorder)
+                .filter(dl -> dl.getTerminal().getBusView().getBus() != null && dl.getTerminal().getBusView().getBus().isInMainSynchronousComponent())
+                .filter(dl -> dl.getExtension(CgmesDanglingLineBoundaryNode.class) == null || !dl.getExtension(CgmesDanglingLineBoundaryNode.class).isHvdc())
```

Comment:
What does the getExtension do?

Reply:
Network components are extendable: additional information can be added as a form of extensions. Here I am checking if there is an extension (called `CgmesDanglingLineBoundaryNode`) and if there is one, if it indicates that the component (here dangling line) is connected to a DC point